src/be_db_oracle_price.erl

```diff
@@ -47,19 +69,46 @@ init(_) ->
     {ok, #state{last_oracle_price=LastPrice}}.
 
 load_block(Conn, _Hash, Block, _Sync, Ledger, State=#state{last_oracle_price=CurrentPrice}) ->
-    case blockchain_ledger_v1:current_oracle_price(Ledger) of
-        {error, Error} ->
-            lager:info("Could not get current oracle price: ~p", [Error]),
-            {ok, State};
-        {ok, CurrentPrice} ->
-            {ok, State};
-        {ok, Price} ->
-            BlockHeight = blockchain_block_v1:height(Block),
-            lager:info("Adjusting oracle at: ~p to: ~p", [BlockHeight, Price]),
-            Queries = [q_insert_oracle_price(BlockHeight, Price)],
-            ok = ?BATCH_QUERY(Conn, Queries),
-            {ok, State#state{last_oracle_price=Price}}
-    end.
+    UpdateOraclePrices =
+        fun(Queries, AccState) ->
+                case blockchain_ledger_v1:current_oracle_price(Ledger) of
+                    {error, Error} ->
+                        lager:info("Could not get current oracle price: ~p", [Error]),
+                        {Queries, AccState};
+                    {ok, CurrentPrice} ->
+                        {Queries, AccState};
+                    {ok, Price} ->
+                        BlockHeight = blockchain_block_v1:height(Block),
+                                        lager:info("Adjusting oracle at: ~p to: ~p", [BlockHeight, Price]),
+                        {[q_insert_oracle_price(BlockHeight, Price) | Queries],
+                         State#state{last_oracle_price=Price}}
+                end
+        end,
+    UpdateOraclePredictions =
+        fun(Queries, AccState) ->
+                NextPrices = blockchain_ledger_v1:next_oracle_prices(Ledger),
```

Comment:
Why not just truncate the table and insert fresh?

Reply:
I already had the adds/delete logic elsewhere too.. but in this case you're definitely right